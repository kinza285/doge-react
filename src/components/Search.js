import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import TextField from '@material-ui/core/TextField/TextField';

const styles = theme => ({
  drawerHeader: {
    ...theme.mixins.toolbar,
    paddingLeft: theme.spacing.unit * 3,
    paddingRight: theme.spacing.unit * 3,
  },
});

class Search extends React.Component {
  static propTypes = {
    classes: PropTypes.objectOf(PropTypes.string).isRequired,
    searchValue: PropTypes.string.isRequired,
  };


  handleSearchChange = (event) => {
    const searchValue = event.target.value;
    // eslint-disable-next-line
    this.props.onSelectSearch(searchValue);
  };

  render() {
    const { classes, searchValue } = this.props;
    return (
      <div className={classes.drawerHeader}>
        <TextField
          fullWidth
          margin="normal"
          placeholder="Search chats..."
          value={searchValue}
          onChange={this.handleSearchChange}
          data={this.state}
        />
      </div>
    );
  }
}
export default withStyles(styles)(Search);
