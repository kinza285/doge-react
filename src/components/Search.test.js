/* eslint-env jest */
import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Search from './Search';

const mockProps = {
  onClick: jest.fn(),
  searchValue: '',
};

describe('<Search />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Search {...mockProps} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders correctly', () => {
    const tree = renderer.create(<Search{...mockProps} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
