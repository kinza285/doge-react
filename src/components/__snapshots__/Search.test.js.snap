// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Search /> renders correctly 1`] = `
<div
  className="Search-drawerHeader-36"
>
  <div
    aria-describedby={undefined}
    className="MuiFormControl-root-37 MuiFormControl-marginNormal-38 MuiFormControl-fullWidth-40"
    data={null}
  >
    <div
      className="MuiInputBase-root-54 MuiInput-root-41 MuiInput-underline-45 MuiInputBase-fullWidth-63 MuiInput-fullWidth-48 MuiInputBase-formControl-55 MuiInput-formControl-42"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        autoComplete={undefined}
        autoFocus={undefined}
        className="MuiInputBase-input-64 MuiInput-input-49"
        defaultValue={undefined}
        disabled={false}
        id={undefined}
        name={undefined}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={undefined}
        onKeyUp={undefined}
        placeholder="Search chats..."
        readOnly={undefined}
        required={false}
        rows={undefined}
        type="text"
        value=""
      />
    </div>
  </div>
</div>
`;
